src/promptflow/promptflow/core/_flow.py

```diff
@@ -167,3 +183,197 @@ async def invoke(self, inputs: dict, *, connections: dict = None, **kwargs) -> "
     @classmethod
     def _create(cls, data, code, path, **kwargs):
         return cls(code=code, path=path, data=data, **kwargs)
+
+
+class Prompty(FlowBase):
+    """A Prompt represents a non-dag flow, which uses prompty file to define the flow.
+    The prompty file is divided into two parts, the first part is in YAML format and contains connection
+    and model information. The second part is the prompt template.
+
+    Prompty example:
+    .. code-block:: yaml
+
+        ---
+        name: Hello Prompty
+        description: A basic prompt
+        api: chat
+        connection: connection_name
+        parameters:
+          deployment_name: gpt-35-turbo
+          max_tokens: 128
+          temperature: 0.2
+        inputs:
+          text:
+            type: string
+        ---
+        system:
+        Write a simple {{text}} program that displays the greeting message.
+
+    Prompty as function example:
+    .. code-block:: python
+
+        from promptflow.core import Prompty
+        prompty = Prompty.load(source="path/to/prompty.prompty")
+        result = prompty(input_a=1, input_b=2)
+
+    """
+
+    def __init__(self, path: Union[str, PathLike], **kwargs):
```

Comment:
The init class should have args like: parameters, api, connection?

Reply:
add those params